packages/core/src/node/environment-utils.ts

```diff
@@ -0,0 +1,74 @@
+/********************************************************************************
+ * Copyright (C) 2021 Ericsson and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { injectable } from 'inversify';
+
+export interface PlatformOptions {
+    platform?: NodeJS.Platform
+}
+
+export interface MergeProcessEnvOptions extends PlatformOptions { }
+
+export interface NormalizeEnvOptions extends PlatformOptions { }
+
+@injectable()
+export class EnvironmentUtils {
+
+    /**
+     * Merge a given record of environment variables with the process environment variables.
+     * Empty string values will not be included in the final env.
+     * @param env desired environment to merge with `process.env`.
+     *
+     * @returns a normalized merged record of valid environment variables.
+     */
+    mergeProcessEnv(env: Record<string, string | null>, options: MergeProcessEnvOptions = {}): Record<string, string> {
+        env = this.normalizeEnv(env, { platform: options.platform });
+        // eslint-disable-next-line no-null/no-null
+        const mergedEnv: Record<string, string> = Object.create(null);
+        for (const [key, value] of Object.entries(this.normalizeEnv(process.env, { platform: options.platform }))) {
+            // Ignore keys from `process.env` that are overridden in `env`. Accept only non-empty strings.
+            if (!(key in env) && value) { mergedEnv[key] = value; }
+        }
+        for (const [key, value] of Object.entries(env)) {
```

Comment:
Are we happy not normalizing these, since they come from the user in any case?

Reply:
I normalized `env` first thing in `mergeProcessEnv`.